automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -490,23 +524,149 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
   }
 
   def keepLongestIdentifier(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {
-      // used to avoid identifiers like R ( t ) being found as separate R, t, R(t, and so on
-      val maxInGroup = new ArrayBuffer[Mention]()
-      val groupedBySent = mentions.groupBy(_.sentence)
-      for (gbs <- groupedBySent) {
-        val groupedByIntervalOverlap = groupByTokenOverlap(gbs._2)
-        for (item <- groupedByIntervalOverlap) {
-          val longest = item._2.maxBy(_.tokenInterval.length)
-          maxInGroup.append(longest)
+    // used to avoid identifiers like R ( t ) being found as separate R, t, R(t, and so on
+    val maxInGroup = new ArrayBuffer[Mention]()
+    val groupedBySent = mentions.groupBy(_.sentence)
+    for (gbs <- groupedBySent) {
+      val groupedByIntervalOverlap = groupByTokenOverlap(gbs._2)
+      for (item <- groupedByIntervalOverlap) {
+        val longest = item._2.maxBy(_.tokenInterval.length)
+        maxInGroup.append(longest)
+      }
+    }
+    maxInGroup.distinct
+  }
+
+
+  // sentence for debugging Bearing in mind , we set our default setting to the refugee move speed is equal to 200 km per day and the awareness of surrounding is 1 link .
```

Comment:
@alicekwak .distinct won't work here because they are not identical mentions :( I have a method for doing distinct by text. I will see if I can use it for param settings.